addons/token-exchange/agent_mirrorpeer_controller.go

```diff
@@ -225,42 +227,42 @@ func (r *MirrorPeerReconciler) createVolumeReplicationClass(ctx context.Context,
 		klog.Error(err, "Failed to get current storage cluster ref")
 		return err
 	}
-	params := make(map[string]string)
-	params[MirroringModeKey] = string(mp.Spec.Mode)
-	params[SchedulingIntervalKey] = mp.Spec.SchedulingInterval
-	params[ReplicationSecretNameKey] = mp.Spec.ReplicationSecretName
-	params[ReplicationSecretNamespaceKey] = scr.Namespace
-	vrc := &replicationv1alpha1.VolumeReplicationClass{
-		ObjectMeta: metav1.ObjectMeta{
-			Name: "odf-rbd-volumereplicationclass",
-		},
-		Spec: replicationv1alpha1.VolumeReplicationClassSpec{
+
+	for _, interval := range mp.Spec.SchedulingInterval {
+		params := make(map[string]string)
+		params[MirroringModeKey] = DefaultMirroringMode
+		params[SchedulingIntervalKey] = interval
+		params[ReplicationSecretNameKey] = ReplicationSecretName
+		params[ReplicationSecretNamespaceKey] = scr.Namespace
+		vrcName := fmt.Sprintf(VolumeReplicationClassNameTemplate, common.FnvHash(interval))
+		found := &replicationv1alpha1.VolumeReplicationClass{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: vrcName,
+			},
+		}
+		err = r.SpokeClient.Get(ctx, types.NamespacedName{
+			Name: found.Name,
+		}, found)
+
+		if err != nil && !errors.IsNotFound(err) {
+			return err
+		}
```

Comment:
return all errors after you have gone through all the intervals. don't block other VRC due to 1 failure.

Reply:
done